src/flb_record_accessor.c

```diff
@@ -453,10 +453,20 @@ static flb_sds_t ra_translate_keymap(struct flb_ra_parser *rp, flb_sds_t buf,
     }
     else if (v->type == FLB_RA_INT) {
         len = snprintf(str, sizeof(str) - 1, "%" PRId64, v->val.i64);
+        /* We need to check size is not above str length */
+        if (len >= 32) {
+            *found = FLB_FALSE;
+            return buf;
+        }
         tmp = flb_sds_cat(buf, str, len);
     }
     else if (v->type == FLB_RA_FLOAT) {
         len = snprintf(str, sizeof(str) - 1, "%f", v->val.f64);
+        /* We need to check size is not above str length */
+        if (len >= 32) {
```

Comment:
Ditto

Reply:
In this case, I think `*found` should be FLB_TRUE and call `flb_sds_cat(buf, str, sizeof(str)-1);`

It causes the loss of precision, but I think it is better to return the value.

Reply:
How about the current one? I updated it with your advice on `flb_sds_cat(buf, str, sizeof(str)-1);` This fixes the bug.

Reply:
Thank you for fixing. LGTM.